### ISSUE TITLE
Add versioning to documentation / don't push documentation of master branch

### ISSUE THREAD

**Issue by jakezatecky** [serverless/serverless#3949]
# This is a Suggestion

## Description

The main website has features documented that are not yet available in the latest release of Severless. This makes it to where developing against the documentation may result in unexplained errors or behavior.

In particular, the documentation published on the main website references that [JavaScript files can be read into a `serverless.yaml` configuration without a named export](https://serverless.com/framework/docs/providers/aws/guide/variables/#reference-variables-in-javascript-files):

> References can be either named or **unnamed exports**. ... For an **unnamed export** you'd write `${file(../myFile.js)}`.

I tried to make use of this feature, but kept getting an error referring to the fact that `.split('.')` was being applied on an undefined object. Eventually, I realized that Serverless was [making use of the colon (`:`) syntax](https://github.com/serverless/serverless/blob/v1.17.0/lib/classes/Variables.js#L234-L235) mentioned on the same documentation page, but not being listed as required:

```
let jsModule = variableString.split(':')[1];
jsModule = jsModule.split('.')[0];
```

Of course, the latest version of this code on `master` supports what is documented on the main website. Nevertheless, the latest version the user is capable of utilizing, v1.17.0, does not have support for this. **The problem is that there is no indication on the main website that the documentation reflects as-of-yet unreleased features. In fact, the documentation says it is for version 1.16.0, which is also incorrect.**

## Additional Data

* ***Serverless Framework Version you're using***: v1.17.0 (latest)
* ***Operating System***: N/A
* ***Stack Trace***: N/A
* ***Provider Error messages***: N/A

**Comment by brianneisler** [comment]
@jakezatecky I think this may actually be a bug rather than a documentation issue. There was some work done a little while ago to make all variables async and it's possible this feature was lost during that work. 

@pmuens @eahefnawy any idea why what happened here? Seems like something that needs to be covered in a test if it's not.

**Comment by jakezatecky** [comment]
For what it's worth, I don't think this was a feature that was lost and added back. It looks like the support of unnamed exports was added in #3841/#3842, which was JUST after the release of v1.17.0. But I haven't been using this project for more than a month, so I could be totally wrong.

**Comment by pmuens** [comment]
Hey @jakezatecky thanks for opening and thanks @brianneisler for jumping in 👍 

It looks like this feature will be shipped with v1.18 since it was merged 13 days ago (see the corresponding PR here: https://github.com/serverless/serverless/pull/3842).

This issue is related to https://github.com/serverless/site/issues/104

/cc @DavidWells

**Comment by pmuens** [comment]
Closing since we've added support for this recently.
